File: gaia-sdk-javascript/src/graphql/response/type/DeletedSkillImpulse.ts

import {TenantKeyOne} from "./TenantKeyOne";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Impulse which indicates the result of a delete Skill impulse
*/
export interface DeletedSkillImpulse {
    id?:Uuid, 
    data?:TenantKeyOne
}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/DeletedSkillImpulse.py

from gaia_sdk.graphql.response.type.KeyOne import KeyOne

from typing import List
Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool
from gaia_sdk.graphql.request.enumeration.RuntimeState import RuntimeState
from gaia_sdk.graphql.request.enumeration.SkillState import SkillState
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField

class DeletedSkillImpulse:
    """
    Impulse which indicates the result of a delete Skill impulse
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        return self.dictionary == other.dictionary

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    @property
    def data(self) -> KeyOne:
        return KeyOne(self.dictionary.get("data"))
